### PR TITLE
fix can change password check in case of encryption is enabled

### LINKED ISSUE #11212: Upgrade 13->14: master key disabled
Upgraded from NC 13.0.6 to 14

I cannot change the passwords of my users any more.

Also a message pops up saying “Password change is disabled because the master key is disabled.”

What can i do without screwing up the file encryption?

### PR DESCRIPTION
**Pull request by schiessle**
Admin should _not_ be able to change password when:
- if an encryption module is loaded and it uses per-user keys
- if encryption is enabled but no encryption modules are loaded

Admin should be able to change the password when:
- no encryption module is loaded and encryption is disabled
- encryption module is loaded but it doesn't require per user keys

fix https://github.com/nextcloud/server/issues/11212

### CODE REVIEW

File: settings/Controller/UsersController.php [hunk 1]
```diff
@@ -166,22 +167,31 @@ public function usersList() {
...
-		$useMasterKey = $this->config->getAppValue('encryption', 'useMasterKey', true);
+		try {
+			$noUserSpecificEncryptionKeys =!$this->encryptionManager->getEncryptionModule()->needDetailedAccessList();
+			$isEncryptionModuleLoaded = true;
+		} catch (ModuleDoesNotExistsException $e) {
+			$noUserSpecificEncryptionKeys = true;
+			$isEncryptionModuleLoaded = false;
+		}
+
 		// If masterKey enabled, then you can change password. This is to avoid data loss!
```
Comment (skjnldsv) [inline settings/Controller/UsersController.php]: Are the comments still accurate?

**Comment by schiessle** [comment]
This makes it easier for people to disable encryption completely and be able to change passwords again as a admin. Also for example, if they just enabled the "default encryption module" app by accident but never really turned on server side encryption.

Of course this carries one (small) risk:
You could imagine a installation where encryption was enabled and files are encrypted with per user keys. Now the admin disables both SSE and the default encryption modules, changes passwords and now realize that they have to enable the default encryption module again to let the user still decrypt old encrypted files. In this case the user can still adjust the private key password in the personal settings if they remember the old password so maybe this is acceptable.

File: settings/Controller/UsersController.php (hunk 1 above)
Comment (schiessle) [reply settings/Controller/UsersController.php]: more or less yes, but let me update the comment and add some more details

Comment (schiessle) [reply settings/Controller/UsersController.php]: done :slightly_smiling_face:

**Comment by schiessle** [comment]
I will move this test to a separate function so that we can write some tests for it...

**Review by skjnldsv — APPROVED** [review]

**Review by rullzer — APPROVED** [review]

**Comment by laurentiu2** [comment]
Does not work for me. I updated from 13.0.8 to 14.0.5 after this was merged and the issue is still there. 

In 13.0.8 I have the master key enabled and as admin I can change the user's password. In 14.0.5 the field for master key is gone.

**Comment by jkoopmann** [comment]
Is it possible this came back with 15.0.7? I have not touched my encryption settings for years and now with 15.0.7 (maybe started with 15.0.x) I suddenly get the warning/error I cannot change passwords:

./occ encryption:status
  - enabled: false
  - defaultModule: OC_DEFAULT_MODULE

**Comment by Jockelchen** [comment]
I've got a more disturbing question but related.
I cant change my users Password since upgrading to nextcloud 16. I can change the Masterkey, but it doesn't get recognized. 

occ encryption:status
  - enabled: true
  - defaultModule: OC_DEFAULT_MODULE

how can i fix this?

**Comment by Jockelchen** [comment]
So, how can i fix this?

**Pull request merged**